### PR TITLE
teach dnx about dotnet, and unteach it about netportable50

### PR DESCRIPTION
**Pull request by anurse**
`lib/dotnet/Foo.dll` ==> `.NETPlatform, Version=v5.0`

/cc @emgarten @yishaigalatzer @muratg @troydai @davidfowl @bricelam @ericstj

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ public static class VersionUtility
         public static readonly string AspNetCoreFrameworkIdentifier = "Asp.NetCore";
         public static readonly string DnxCoreFrameworkIdentifier = "DNXCore";
         public static readonly string PortableFrameworkIdentifier = ".NETPortable";
```
Comment (muratg) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: Should we remove this?

Comment (anurse) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: No, it is still used for profile-based portable (`portable-net45+win8`, etc.)

**Comment by anurse** [comment]
Travis failure seems to be unrelated

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs [hunk 2]
```diff
@@ -251,7 +246,7 @@ public static FrameworkName ParseFrameworkName(string frameworkName)
                     return UnsupportedFrameworkName;
                 }
 
-                version = _emptyVersion;
+                version = identifierPart.Equals(NetPlatformFrameworkIdentifier) ? new Version(5, 0) : _emptyVersion;
```
Comment (anurse) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: This is turning out to be a bit weird. So I'm going to drop it. The Version Number for "dotnet" will be "0.0". We can display it as "5.0" in the places we want to, if we want to.

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: I just sent out PRs to remove all these from our repos...

**Comment by bricelam** [comment]
Will the defines flag be `DOTNET` or `DOTNET00`?

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs (hunk 2 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: @emgarten came through with a clean way to do this in the new nuget code, so we can keep it! Ignore my previous comment :).

**Comment by anurse** [comment]
Good question, I'll double check but it should be DOTNET since I think we take the short name nuget gives us.

**Comment by anurse** [comment]
To answer your question @bricelam , it will be whatever you typed in "frameworks", but uppercase. So if you typed "dotnet00" it would be DOTNET00. This is true for any framework.

**Comment by bricelam** [comment]
Cool.

**Pull request closed without merging**